website/content/docs/pipelines/index.mdx

```diff
@@ -0,0 +1,438 @@
+          prune = true
+        }
+      }
+    }
+  }
+}
+```
+
+~> **NOTE**: Currently, this Tech Preview only allows for 1 level of nesting when defining an embedded pipeline. However, Pipelines can also be used as a reference within a step.
+
+### Reference Pipelines As Steps
+
+You can reference other Pipelines within a Project via a `pipeline` step:
+
+```hcl
+project = "my-pipeline-project"
+
+pipeline "simple-nested" {
+  step "build" {
+    use "build" {}
+  }
+
+  step "my-deploy" {
+    #  "pipeline" is a special "built-in" step that references another pipeline
+    use "pipeline" {
+      project = "my-pipeline-project"
+      name    = "deploy"
+    }
+  }
+}
+
+pipeline "deploy" {
+  step "deploy" {
+    use "deploy" {}
+  }
+  step "release" {
+    use "release" {
+      prune = true
+    }
+  }
+}
+```
+
+### Pipelines Across Workspaces
+
+Pipelines can be defined and scoped to a Workspace for deploying in various environments.
+When you use the pipeline run CLI, you can run a pipeline in a different workspace
+with the `-workspace` flag.
+
+```shell
+$ workspace pipeline run builder -workspace=dev
+```
+
+#### Workspace Scoped Steps
+
+You can override the environment that a Pipeline step runs in by specifying a
+`workspace` param for any step in your Pipeline definition. For example,
+this pipelines "build-prod" step will run in the production workspace.
+
+Notice that in the app definition, it contains two build stanzas, one of which
+is _workspace scoped_. That means when the "build-prod" step runs, it will
+use that workspace scoped build plugin pack rather than the default "docker" plugin.
+
+```hcl
+// waypoint.hcl
+pipeline "builder" {
+  step "build-dev" {
+    use "build" {}
+  }
+  step "build-prod" {
+    workspace = "prod"
+    use "build" {}
+  }
+  step "deploy-dev" {
+    use "deploy" {
+      release = false
+    }
+  }
+}
+
+// waypoint.hcl
+app "tetris" {
+  build {
+    use "docker" {}
+
+    workspace "prod" {
+      use "pack" {}
+    }
+  }
+  deploy {
+    use "docker" {}
+
+    workspace "prod" {
+      use "kubernetes" {}
+    }
+  }
+}
+```
+
+You can also use this to set a workspace for an embedded pipeline:
+
+```hcl
+pipeline "builder" {
+  step "build" {
+    use "build" {}
+  }
+  step "deploy-test" {
+    workspace = "test"
+    pipeline "deploy-test-env" {
+      step "build" {
+        // test env uses docker-pull
+        use "build" {}
+      }
+      step "deploy" {
+        use "deploy" {}
+      }
+      // more steps here...
+    }
+  }
+  step "deploy-prod" {
+    workspace = "prod"
+    pipeline "deploy-prod-env" {
+      step "build" {
+        // prod env uses docker-pull
+        use "build" {}
+      }
+      step "deploy" {
+        use "deploy" {}
+      }
+      // more steps here...
+    }
+  }
+}
+```
+
+## Examples
+
+Below is an example that redefines the Up operation to be a pipeline with three
+steps for a build, deploy, and release. This example could be used as a starting
```

Comment:
```suggestion
steps for a build, deploy, and release workflow. This example could be used as a starting
```

Reply:
"Workflow" may be an overloaded term, so I'm ok if we don't use it. But yea either 
"steps for build, deploy, and release" 
or 
"steps for a build, deploy, and release <something>"

Reply:
I think it makes sense! I'll go for it